resources/2024-02-output-resources.md

```diff
@@ -0,0 +1,210 @@
+  localId?: string;
+
+  @doc("The UCP resource ID of the underlying resource.")
+  id?: string;
+
+  @doc("Determines whether Radius manages the lifecycle of the underlying resource.")
+  radiusManaged?: boolean;
+
+  @doc("Determines whether the resource was provisioned successfully")
+  provisioningStatus: ProvisioningStatus
+}
+
+@doc("Provisioning Status of Output Resource")
+enum ProvisioningStatus {
+  @doc("The resource is successfully provisioned")
+  Succeeded,
+
+  @doc("The resource provisioning failed")
+  Failed,
+}
+`typespec``
+
+***Example with proposed change***
+rad resource list containers -a allinoneapp -o json                
+[
+  {
+    "id": "/planes/radius/local/resourcegroups/randomgroup/providers/Applications.Core/containers/allinonecontainer",
+    "location": "global",
+    "name": "allinonecontainer",
+    "properties": {
+      "application": "/planes/radius/local/resourcegroups/randomgroup/providers/Applications.Core/applications/allinoneapp",
+      :
+      "provisioningState": "Succeeded",  ### or "Failed/ Non-Success status" 
+      "status": {
+        "outputResources": [
+          {
+            "id": "/planes/kubernetes/local/namespaces/allinoneenv-allinoneapp/providers/core/ServiceAccount/allinonecontainer",
+            "localId": "ServiceAccount"
+            "provisioningState": "Succeeded"
+          },
+          {
+            "id": "/planes/kubernetes/local/namespaces/allinoneenv-allinoneapp/providers/rbac.authorization.k8s.io/Role/allinonecontainer",
+            "localId": "KubernetesRole"
+            "provisioningState": "Succeeded"
+          },
+          {
+            "id": "/planes/kubernetes/local/namespaces/allinoneenv-allinoneapp/providers/rbac.authorization.k8s.io/RoleBinding/allinonecontainer",
+            "localId": "KubernetesRoleBinding"
+            "provisioningState": "Succeeded"
+          },
+          {
+            "id": "/planes/kubernetes/local/namespaces/allinoneenv-allinoneapp/providers/apps/Deployment/allinonecontainer",
+            "localId": "Deployment"
+            "provisioningState": "Failed"
+          },
+          {
+            "id": "/planes/kubernetes/local/namespaces/allinoneenv-allinoneapp/providers/core/Service/allinonecontainer",
+            "localId": "Service"
+            "provisioningState": "Succeeded"
+          }
+        ]
+      }
+      :
+]
+```
+
+
+## Alternatives considered
+
+NA
+
+## Test plan
+
+1. Updating conversion unit tests to make sure the new attribute is handled correctly
+2. Updating some API unit tests to see if the information is being populated on the resource level.
+
+## Monitoring
+
+N/A
+
+## Development plan
+
+There is an in-progress PR which brings in the proposed changes as part of fix for rad app delete. 
```

Comment:
There would be some changes needed in the delete logic as well, right?

Reply:
Yes. https://github.com/radius-project/radius/pull/7148 
The PR needs to be updated based on the feedback I receive here, but it tests the idea out and worked in case of container deployment failures  followed by a rad app delete.